.jenkinsci-new/builders/x64-mac-build-steps.groovy

```diff
@@ -0,0 +1,100 @@
+        mkdir -p /var/jenkins/${scmVars.GIT_COMMIT}-${BUILD_NUMBER}; \
+        initdb -D /var/jenkins/${scmVars.GIT_COMMIT}-${BUILD_NUMBER}/ -U ${IROHA_POSTGRES_USER} --pwfile=<(echo ${IROHA_POSTGRES_PASSWORD}); \
+        pg_ctl -D /var/jenkins/${scmVars.GIT_COMMIT}-${BUILD_NUMBER}/ -o '-p 5433 -c max_prepared_transactions=100' -l /var/jenkins/${scmVars.GIT_COMMIT}-${BUILD_NUMBER}/events.log start; \
+        psql -h localhost -d postgres -p 5433 -U ${IROHA_POSTGRES_USER} --file=<(echo create database ${IROHA_POSTGRES_USER};)
+      """
+      sh "cd build; IROHA_POSTGRES_HOST=localhost IROHA_POSTGRES_PORT=5433 ctest --output-on-failure --no-compress-output --tests-regex '${testList}'  --test-action Test || true"
+
+      sh 'python .jenkinsci-new/helpers/platform_tag.py "Darwin \$(uname -m)" \$(ls build/Testing/*/Test.xml)'
+      // Mark build as UNSTABLE if there are any failed tests (threshold <100%)
+      xunit testTimeMargin: '3000', thresholdMode: 2, thresholds: [passed(unstableThreshold: '100')], \
+        tools: [CTest(deleteOutputFiles: true, failIfNotNew: false, \
+        pattern: 'build/Testing/**/Test.xml', skipNoTestFiles: false, stopProcessingIfError: true)]
+
+      sh """
+        pg_ctl -D /var/jenkins/${scmVars.GIT_COMMIT}-${BUILD_NUMBER}/ stop && \
+        rm -rf /var/jenkins/${scmVars.GIT_COMMIT}-${BUILD_NUMBER}/
+      """
+  }
+}
+
+def buildSteps( int parallelism, List compilerVersions, String build_type, boolean coverage, boolean testing, String testList, boolean packagebuild, List environment) {
+  withEnv(environment) {
+    scmVars = checkout scm
+    def build = load '.jenkinsci-new/build.groovy'
+    def vars = load ".jenkinsci-new/utils/vars.groovy"
+    def utils = load ".jenkinsci-new/utils/utils.groovy"
+    buildDir = 'build'
+    compilers = vars.compilerMapping()
+    cmakeBooleanOption = [ (true): 'ON', (false): 'OFF' ]
+        cmakeBuildOptions = ""
+
+    if (packagebuild){
+      cmakeBuildOptions = " --target package "
+    }
+
+    utils.ccacheSetup(5)
+
+    for (compiler in compilerVersions) {
+      stage ("build ${compiler}"){
+        build.cmakeConfigure(buildDir,
+        "-DCMAKE_CXX_COMPILER=${compilers[compiler]['cxx_compiler']} \
+        -DCMAKE_C_COMPILER=${compilers[compiler]['cc_compiler']} \
+        -DCMAKE_BUILD_TYPE=${build_type} \
+        -DCOVERAGE=${cmakeBooleanOption[coverage]} \
+        -DTESTING=${cmakeBooleanOption[testing]} \
+        -DPACKAGE_TGZ=${cmakeBooleanOption[packagebuild]} ")
+
+        build.cmakeBuild(buildDir, cmakeBuildOptions, parallelism)
+      }
+      if (testing) {
+        stage("Test ${compiler}") {
+          coverage ? build.initialCoverage(buildDir) : echo('Skipping initial coverage...')
+          testSteps(scmVars, buildDir, environment, testList)
+          coverage ? build.postCoverage(buildDir, '/usr/local/bin/lcov_cobertura.py') : echo('Skipping post coverage...')
+          // We run coverage one time,in first compiler, it enough
```

Comment:
// We run coverage once, using the first compiler as it is enough

Reply:
Jenkins stores them internally. The report itself is available with each build it was enabled for.